ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java

```diff
@@ -185,6 +188,12 @@ public int monitorExecution() {
         if (wmContext != null) {
           Set<String> desiredCounters = wmContext.getSubscribedCounters();
           TezCounters dagCounters = status.getDAGCounters();
+          // if initial counters exists, merge it with dag counters to get aggregated view
```

Comment:
Can we move this if block to a utility static function `mergeCounters`? It seems it is repeated in a couple of places, may be useful in other cases in the future.

Reply:
Done.